tests/ui/single_component_path_imports.rs

```diff
@@ -19,3 +19,9 @@ fn main() {
     // False positive #5154, shouldn't trigger lint.
     m!();
 }
+
+mod hello_mod {
+    use regex;
+    #[allow(dead_code)]
+    fn hello_mod() {}
+}
```

Comment:
Thanks for moving this forward!

I think we need to check it for each module. If the codes like below added to this test code, it looks like that the lint isn't emitted at all in this test.
```rust
mod hi_mod {
    use self::regex::{Regex, RegexSet};
    use regex;
    #[allow(dead_code)]
    fn hi_mod() {}
}
```